streamer/src/streamer.rs

```diff
@@ -66,30 +88,44 @@ impl StreamerReceiveStats {
             packet_batches_count: AtomicUsize::default(),
             full_packet_batches_count: AtomicUsize::default(),
             max_channel_len: AtomicUsize::default(),
+            total_stats: Arc::new(Mutex::new(StreamerReceiveStatsTotal::default())),
         }
     }
 
     pub fn report(&self) {
+        let packets_count = self.packets_count.swap(0, Ordering::Relaxed);
+        let packet_batches_count = self.packets_count.swap(0, Ordering::Relaxed);
+        let full_packet_batches_count = self.packets_count.swap(0, Ordering::Relaxed);
+        let max_channel_len = self.packets_count.swap(0, Ordering::Relaxed);
+
+        {
+            let mut stats = self.total_stats.lock().unwrap();
```

Comment:
What are `packet_batches_count`, `full_packet_batches_count` and `max_channel_len`?

Reply:
So this is all kind of messy, but from what I understand:
- `packets_count` here is just number of UDP packets received specifically on TPU sockets.
- `packet_batches_count` is number of "batches" received - Solana will read packets from the socket in batches of up to 64 packets.
- `full_packet_batches_count` - is number of batches that were full (so containing 64 packets).
- `max_channel_len` - after reading from the socket packets are forwarded to unbounded channel. So this value is is max length of this channel (across all current measurements), measured each time before forwarding new packets. So this will basically be a max value over 1 second since then the stats are reset 🤷

Reply:
Can you add those as the `HELP` line in the exported metrics?

Reply:
> So this will basically be a max value over 1 second since then the stats are reset

I don’t know how it measures this, but if the channel is backed by something similar to a `Vec` (I think it might be a deque in practice, but the same applies), and the max measures the capacity of the vec/deque, then it would persist even if the metrics are reset, unless they `shrink_to_fit`. But this is pure speculation, I haven’t looked into how it is implemented.